azurelinuxagent/common/event.py

```diff
@@ -256,7 +256,7 @@ def add_event(self,
                   is_internal=False,
                   log_event=True):
 
-        if not is_success or log_event:
+        if not is_success and log_event:
```

Comment:
Consider adding parentheses to make precedence clear without requiring the reader to look up whether `not` or `and` binds more tightly.

Reply:
Updated